syntax = "proto3";
package raft_rpc;

message rpc_Entry {
    int32 term = 1;
    string msg = 2;
}

message AppendEntryRpc {
    int32 term = 1;
    int32 index = 2;
    int32 prelog_term = 3;
    int32 prelog_index = 4;
    int32 commit_index = 5;
    repeated rpc_Entry entry = 6;
}


message Resp_AppendEntryRPC {
    bool ok = 1;
    int32 term = 2;
}


message RequestVoteRpc {
    int32 term = 1;
    int32 index = 2;
}


message Resp_RequestVoteRpc {
    bool ok = 1;
    int32 term = 2;
}

